common/str.h

```diff
@@ -224,6 +235,23 @@ class String {
 	uint hash() const;
 
 	/**
+	 * Various replace functions to suit different parameters.
+	 * 
+	 * For now, they are unable to change the size of the string.
+	 * Thus, they do not insert, they replace the characters.
+	 */
+	// Replace 'count' bits, starting from 'pos' with str[pos, pos+count)
+	void replace(uint32 pos, uint32 count, const String& str);
+	// The same as above, but accepts a C-like array of characters	
+	void replace(uint32 pos, uint32 count, const char* str);
+	// Replace the space between the iterators with str[begin, end)
+	void replace(iterator begin, iterator end, const String& str);
+	// Replace _str[pos_ori, pos_ori+count_ori) with 
+	// str._str[pos_dest, pos_dest+count_dest)
```

Comment:
This is also slightly inaccurate. count_dest is never used in the implementation